part-iv-advanced-tuning/metrics-and-monitoring.md

```diff
@@ -0,0 +1,108 @@
+# Metrics & Monitoring
+
+Important in any application is the ability to monitor and measure what is going on. 
+Especially in a location transparent environment like an Axon application it is very important to be able to trace your message and check the ingestion rate of it.
+
+## Metrics
+
+Interesting metrics in a message centric system come in several forms and flavors, like count, capacity and latency for example.  
+The Axon Framework allows you to retrieve such measurements through the use of the `axon-metrics` module.
+With this module you can registers a number of `MessageMonitor` implementations to your messaging components, like the [`CommandBus`](../part-iii-infrastructure-components/command-dispatching.md#the-command-bus), [`EventBus`](../part-iii-infrastructure-components/event-processing.md#event-bus), [`QueryBus`](../part-iii-infrastructure-components/query-processing.md#query-bus) and [`EventProcessors`](../part-iii-infrastructure-components/event-processing.md#event-processors).
+
+Internally, the `axon-metrics` module uses the [Dropwizard Metrics](https://metrics.dropwizard.io/) for registering the measurements correctly.
+That thus means that `MessageMonitors` are registered against the Dropwizard `MetricRegistry`.
+The following monitor implementations are currently provided:
+1. `CapacityMonitor` - Calculates capacity by tracking, within a configurable time window, the average message processing time and multiplying that by the amount of messages processed.
+2. `EventProcessorLatencyMonitor` - Measures the difference in message timestamps between the last ingested and the last processed event message.
+3. `MessageCountingMonitor` - Counts the number of ingested, successful, failed, ignored and processed messages.
+4. `MessageTimerMonitor` - Keeps a timer for all successful, failed and ignored messages, as well as an overall timer for all three combined.
+5. `PayloadTypeMessageMonitorWrapper` - A special `MessageMonitor` implementation which allows setting a monitor per message type instead of per message publishing/handling component. 
```

Comment:
Why is this one special?

Reply:
Because it does not measure anything it self, it instantiates new `MessageMonitors` for every new payload type it encounters.